components/feature/intent/src/main/java/mozilla/components/feature/intent/ext/IntentExtensions.kt

```diff
@@ -25,6 +26,14 @@ fun Intent.getSessionId(): String? = getStringExtra(EXTRA_SESSION_ID)
  */
 fun SafeIntent.getSessionId(): String? = getStringExtra(EXTRA_SESSION_ID)
 
+/**
+ * Retrieves [String] Url override from the intent.
+ *
+ * @return The URL override previously added with [putUrlOverride],
+ * or null if no URL was found.
+ */
+fun Intent.getUrlOverride(): String? = getStringExtra(EXTRA_URL_OVERRIDE)
```

Comment:
Personally I think it would be better to have either the manifest or the start url included as an extra, rather than the real URL. We already have some code for embedding a web app manifest into an intent.

Reply:
But we need the real URL to deep link into the PWA. If I for example click on a link for a tweet I want to end up on that tweet inside the PWA and not the start URL.

Reply:
We also need this because we always have to use the same intent URI or otherwise android will create new Activities for each URI.